### PR TITLE
Set up automatic build pipeline for amd64 target

### PR DESCRIPTION
**Pull request by hcho3**
Use CentOS 6 Docker image to build a universal DLR CPU artifact that can run on all amd64 targets.

TODO: support AMD64+CUDA build

### CODE REVIEW

**Comment by hcho3** [comment]
@wweic @zhiics @ziyu-guo Sorry for the noise. This PR is ready for review now. All built artifacts will be uploaded to a private S3 bucket.

**Comment by zhiics** [comment]
@hcho3 Thanks for setting up. It looks good to me.

**Comment by hcho3** [comment]
See example of the build job at https://neo-ai-ci.amazon-ml.com/blue/organizations/jenkins/neo-ai-dlr/detail/PR-28/12/pipeline

File: Jenkinsfile [hunk 1]
```diff
@@ -59,24 +55,19 @@ def checkoutSrcs() {
...
-def buildHelloWorldJob(worker_type) {
-  def nodeReq = "linux && ${worker_type} && unrestricted"
+def AMD64BuildCPU() {
+  def nodeReq = "ubuntu && amd64 && cpu-bare"
+  def dockerTarget = "cpu_bare"
+  def dockerArgs = ""
   node(nodeReq) {
     unstash name: 'srcs'
-    echo "Hello world!"
+    echo "Building univeral artifact for AMD64, CPU-only"
```
Comment (wweic) [inline Jenkinsfile]: `univeral` -> `universal`

File: tests/ci_build/ci_build.sh [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        test -e "$1" && echo "$PWD" && return || \
+        cd .. && upsearch "$1"
+}
+
+# Set up WORKSPACE. Jenkins will set them for you or we pick
+# reasonable defaults if you run it outside of Jenkins.
+WORKSPACE="${WORKSPACE:-${SCRIPT_DIR}/../../}"
+
+# Determine the docker image name
+DOCKER_IMG_NAME="xgb-ci.${CONTAINER_TYPE}"
```
Comment (wweic) [inline tests/ci_build/ci_build.sh]: What's inside this docker image?

Comment (hcho3) [reply tests/ci_build/ci_build.sh]: Should be renamed to `neo-ai-dlr-build`

**Pull request merged**